tests/unit/consensus/test_effect_executor.py

```
@@ -0,0 +1,500 @@
+        with pytest.raises(TypeError, match="Unknown effect type"):
+            await executor.execute([unknown])
+
+
+@pytest.mark.asyncio
+class TestMultipleEffects:
+    async def test_executes_effects_in_order(self):
+        ctx = _make_context()
+        executor = EffectExecutor(ctx)
+        call_order = []
+        ctx.transactions_processor.add_state_timestamp.side_effect = (
+            lambda *a: call_order.append("timestamp")
```

Ruff (0.15.2)

[warning] Unused lambda argument: `a`

(ARG005)